### ISSUE TITLE
the tasks in the DAG should added to workflow automatically

### ISSUE THREAD

**Issue by hirosassa** [argoproj-labs/hera-workflows#187]
Hi, developers

Thank you for developing great Python SDK for Argo workflow!

I would like to discuss on the issue below.
From the example on README
```python
w = Workflow(...)
a = Task(...)
b = Task(...)
c = Task(...)

a >> b >> c
w.add_tasks(a, b, c) 
```
I feel the last line of the example is redundant because the tasks contains in the workflow DAG (in the example, `a >> b >> c`) should added automatically in workflow.
I and my team members often forget adding new task in `w.add_tasks()`.

**Comment by flaviuvadan** [comment]
Hello @hirosassa! Nice to see issue submissions from you again 🚀 Yea, this idea is actually very interesting and I am still conflicted about whether it should be implemented. First, by the nature of Python the implementation of this feature will not be thread safe unless we build something like a context manager, such as [this one](https://github.com/artigraph/artigraph/blob/4011d565532f5d83a8b917df9a1a214c638b2b0c/src/arti/__init__.py#L58). Using a context manager will allow a task to add itself to some abstract container, then that container can obtain the currently instantiated workflow, and add the task from the container to the workflow. Upon submission the workflow is created and the context is emptied. However, I think this may expose some users who launch workflows from servers to potential locks unless there are multiple workers running a server. Second, one of the pieces of feedback I have heard about Hera before is that the explicit nature of graph construction is something that multiple users appreciate. The graph is "clear" so to say, and there's nothing implicit about the execution of a workflow. I guess that something like 
```python
a = Task(...)
b = Task(...)
c = Task(...) 
```
would not imply a sequential execution. Rather, it would imply parallel execution since adding tasks to a workflow means Argo will execute them in parallel if there are no dependencies set via ```a >> b >> c```. Parts of this discussion were covered in [this issue](https://github.com/argoproj-labs/hera-workflows/issues/79), which is still open. Perhaps it's time to revisit the topic. Any thoughts?

**Comment by flaviuvadan** [comment]
https://github.com/argoproj-labs/hera-workflows/pull/209

**Comment by hirosassa** [comment]
Thanks @flaviuvadan noticing me the update! I'm sorry for leaving this issue.
I think this is solve in #209 . 
I'll close this issue.

**Comment by flaviuvadan** [comment]
No worries at all! Please let me know what you think of the context usage 🙂 
```python
with Workflow(...) as w:
    Task(...) >> Task(...)

w.create()
```
